godot-macros/src/derive_godot_class.rs

```diff
@@ -265,51 +265,109 @@ fn make_deref_impl(class_name: &Ident, fields: &Fields) -> TokenStream {
-            let setter = proc_macro2::Literal::from_str(&exported_field.setter).unwrap();
-            let field_type = exported_field.field.ty.clone();
-
-            let ExportHint {
-                hint_type,
-                description,
-            } = exported_field.hint.clone().unwrap_or_else(ExportHint::none);
-
-            // trims '"' and '\' from both ends of the hint description.
-            let description = description.trim_matches(|c| c == '\\' || c == '"');
-
-            quote! {
-                use ::godot::builtin::meta::VariantMetadata;
-
-                let class_name = ::godot::builtin::StringName::from(#class_name::CLASS_NAME);
-                let property_info = ::godot::builtin::meta::PropertyInfo::new(
-                    <#field_type>::variant_type(),
-                    ::godot::builtin::meta::ClassName::of::<#class_name>(),
-                    ::godot::builtin::StringName::from(#name),
-                    ::godot::engine::global::PropertyHint::#hint_type,
-                    GodotString::from(#description),
-                );
-                let property_info_sys = property_info.property_sys();
-
-                let getter_string_name = ::godot::builtin::StringName::from(#getter);
-                let setter_string_name = ::godot::builtin::StringName::from(#setter);
-                unsafe {
-                    ::godot::sys::interface_fn!(classdb_register_extension_class_property)(
-                        ::godot::sys::get_library(),
-                        class_name.string_sys(),
-                        std::ptr::addr_of!(property_info_sys),
-                        setter_string_name.string_sys(),
-                        getter_string_name.string_sys(),
-                    );
+    use std::str::FromStr;
+
+    let mut getter_setter_impls = Vec::new();
+    let mut export_tokens = Vec::with_capacity(fields.exported_fields.len());
+
+    for exported_field in &fields.exported_fields {
+        let field_name = exported_field.field.name.to_string();
+        let field_ident = proc_macro2::Ident::new(&field_name, proc_macro2::Span::call_site());
+        let field_type = exported_field.field.ty.clone();
+
+        let getter_name = exported_field
+            .getter
+            .as_ref()
+            .map(String::to_owned)
+            .unwrap_or_else(|| format!("\"get_{}\"", field_name));
+        let setter_name = exported_field
+            .setter
+            .as_ref()
+            .map(String::to_owned)
+            .unwrap_or_else(|| format!("\"set_{}\"", field_name));
+        let getter_lit = proc_macro2::Literal::from_str(&getter_name).unwrap();
+        let setter_lit = proc_macro2::Literal::from_str(&setter_name).unwrap();
+        let getter_ident =
+            proc_macro2::Ident::new(&strip_quotes(&getter_name), proc_macro2::Span::call_site());
+        let setter_ident =
+            proc_macro2::Ident::new(&strip_quotes(&setter_name), proc_macro2::Span::call_site());
```

Comment:
> Note that you can directly pass strings into quote! and they will be formatted as literals like "string" (the previous code was more complex than necessary, too).

OK, let me remove `strlit` in utils too then, since it does this same thing and is unused anyway.

> For the other part, do we actually need strip_quotes here, or can we just use the ident() utility function? Possibly exported_field.getter should already be populated correctly in the first place, i.e. getting the unquoted string from KvParser directly.

Nope, the value is quoted in the original attribute as well, i.e. `get = "get_my_value"` results in `getter_name == "\"get_my_value\""`. `strip_quotes` was just my first attempt, definitely not final -- we should throw an actual Rust parser at this task. But it seems venial can only parse declarations, not expressions :(

Alternatively, we omit the quotes from the macro syntax and parse it as an identifier. Somehow this is not commonly done (see e.g. the `default` attr in serde), but I don't know why.

Reply:
> OK, let me remove strlit in utils too then, since it does this same thing and is unused anyway.

Sounds good! I think I wrote that before I realized I could use strings directly 😀 

> we should throw an actual Rust parser at this task. But it seems venial can only parse declarations, not expressions :(

Why do we need a Rust parser if all we want is strip quotes? We don't have to support arbitrary function expressions, limiting it to methods for now is completely fine. venial is deliberately kept simple as a proc-macro processor, not a full language parser 🙂 

I think we shouldn't overengineer it. `strip_prefix` + `strip_postfix` + error handling, and `ident()` will also panic if we pass it a weird string. We can always extend it in the future.

> Alternatively, we omit the quotes from the macro syntax and parse it as an identifier. Somehow this is not commonly done (see e.g. the default attr in serde), but I don't know why.

There was some previous discussion in https://github.com/godot-rust/gdext/pull/31#issuecomment-1331678068. I also start to believe quotes might be better, because:
* ecosystem convention
* possibly allows more complex expressions (not sure if relevant here)
* visual separation between key and value, even more with syntax highlighting

I'm happy to hear more input on this topic though.

Reply:
> Why do we need a Rust parser if all we want is strip quotes?

I'm pretty sure that naive string splitting can break on raw strings and escape sequences. I'll just document for now that getter/setter identifiers should be ASCII and not be Rust keywords.

Reply:
I think limiting to ASCII is a fair assumption to start with. If people really need Unicode support, I'm happy to accept PRs with a concrete use case 🙂